### PR TITLE
Remove Legacy dependency from Microsoft.ML.Benchmarks

### PR DESCRIPTION
**Pull request by najeeb-kazmi**
Part of the work in PR #1971 

This PR removes dependency on Legacy code from Microsoft.ML.Benchmarks.

`LegacyPredictionEngineBench.cs` was written to mimic `PredictionEngineBench.cs` using Legacy API after the fact, so just deleting it here.

`StochasticDualCoordinateAscentClassifierBench.cs` is rewritten to use new API and remove all Legacy code.

### CODE REVIEW

File: test/Microsoft.ML.Benchmarks/StochasticDualCoordinateAscentClassifierBench.cs [hunk 1]
```diff
@@ -123,39 +132,34 @@ public void TrainSentiment()
...
+                    },
+                    hasHeader: true
+                );
 
-            _batches = new IrisData[_batchSizes.Length][];
-            for (int i = 0; i < _batches.Length; i++)
-            {
-                var batch = new IrisData[_batchSizes[i]];
-                _batches[i] = batch;
-                for (int bi = 0; bi < batch.Length; bi++)
```
Comment (najeeb-kazmi) [inline test/Microsoft.ML.Benchmarks/StochasticDualCoordinateAscentClassifierBench.cs]: `BatchPredictionEngine` is not part of new API, so removing any code related to batch predictions.

File: test/Microsoft.ML.Benchmarks/StochasticDualCoordinateAscentClassifierBench.cs [hunk 2]
```diff
@@ -123,39 +132,34 @@ public void TrainSentiment()
...
-        public float[] PredictIris() => _trainedModel.Predict(_example).PredictedLabels;
-
-        [Benchmark]
-        public void PredictIrisBatchOf1() => Consume(_trainedModel.Predict(_batches[0]));
-
-        [Benchmark]
-        public void PredictIrisBatchOf2() => Consume(_trainedModel.Predict(_batches[1]));
-
-        [Benchmark]
-        public void PredictIrisBatchOf5() => Consume(_trainedModel.Predict(_batches[2]));
```
Comment (najeeb-kazmi) [inline test/Microsoft.ML.Benchmarks/StochasticDualCoordinateAscentClassifierBench.cs]: `BatchPredictionEngine` is not part of new API, so removing any code related to batch predictions.

File: test/Microsoft.ML.Benchmarks/StochasticDualCoordinateAscentClassifierBench.cs [hunk 3]
```diff
@@ -31,31 +29,42 @@ public class StochasticDualCoordinateAscentClassifierBench : WithExtraMetrics
             PetalWidth = 5.1f,
         };
 
-        private Legacy.PredictionModel<IrisData, IrisPrediction> _trainedModel;
-        private IrisData[][] _batches;
```
Comment (najeeb-kazmi) [inline test/Microsoft.ML.Benchmarks/StochasticDualCoordinateAscentClassifierBench.cs]: >private IrisData[][] _batches; [](start = 8, length = 30)

`BatchPredictionEngine` is not part of new API, so removing any code related to batch predictions.

File: test/Microsoft.ML.Benchmarks/StochasticDualCoordinateAscentClassifierBench.cs [hunk 4]
```diff
@@ -22,7 +21,6 @@ public class StochasticDualCoordinateAscentClassifierBench : WithExtraMetrics
         private readonly string _sentimentDataPath = Program.GetInvariantCultureDataPath("wikipedia-detox-250-line-data.tsv");
         private readonly Consumer _consumer = new Consumer(); // BenchmarkDotNet utility type used to prevent dead code elimination
 
-        private readonly int[] _batchSizes = new int[] { 1, 2, 5 };
```
Comment (najeeb-kazmi) [inline test/Microsoft.ML.Benchmarks/StochasticDualCoordinateAscentClassifierBench.cs]: >private readonly int[] _batchSizes = new int[] { 1, 2, 5 }; [](start = 8, length = 59)

`BatchPredictionEngine` is not part of new API, so removing any code related to batch predictions.

File: test/Microsoft.ML.Benchmarks/StochasticDualCoordinateAscentClassifierBench.cs [hunk 5]
```diff
@@ -31,31 +30,43 @@ public class StochasticDualCoordinateAscentClassifierBench : WithExtraMetrics
...
         [Benchmark]
-        public Legacy.PredictionModel<IrisData, IrisPrediction> TrainIris() => Train(_dataPath);
+        public TransformerChain<MulticlassPredictionTransformer<MulticlassLogisticRegressionModelParameters>> TrainIris() => Train(_dataPath);
 
-        private Legacy.PredictionModel<IrisData, IrisPrediction> Train(string dataPath)
+        private TransformerChain<MulticlassPredictionTransformer<MulticlassLogisticRegressionModelParameters>> Train(string dataPath)
         {
-            var pipeline = new Legacy.LearningPipeline();
+            var env = new MLContext(seed: 1, conc: 1);
+            var reader = new TextLoader(env,
```
Comment (artidoro) [inline test/Microsoft.ML.Benchmarks/StochasticDualCoordinateAscentClassifierBench.cs]: Can you use a single mlContext for both training and predicting? Would require defining it as a global variable, and using it in both training and prediction code.

**Pull request merged**